NitroxReloader/Reloader.cs

```diff
@@ -0,0 +1,244 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Reflection.Emit;
+using System.Runtime.CompilerServices;
+
+// Reloader based on http://github.com/pardeike/Reloader
+// Modified to remove RimWorld dependency, and cleanups.
+
+namespace NitroxReloader
+{
+    [AttributeUsage(AttributeTargets.Method)]
+    public class ReloadableMethod : Attribute
+    {
+    }
+
+    public class Reloader
```

Comment:
I would consider refactoring the classes into individual files to make it easier to maintain and evolve.